code/src/context/ModalContext.tsx

```diff
@@ -0,0 +1,57 @@
+import {
+  ReactElement,
+  ReactNode,
+  createContext,
+  useContext,
+  useRef,
+  useState
+} from "react";
+import Dialog from "@mui/material/Dialog";
+import AppBox from "@/components/app-box/AppBox";
+
+type ModalContextType = {
+  handleOpenModal: (component: ReactElement) => void;
+  handleCloseModal: () => void;
+};
+
+type ModalProviderProps = { children: ReactNode };
+
+const ModalContext = createContext<ModalContextType | null>(null);
+
+const ModalProvider = ({ children }: ModalProviderProps) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const modalRef = useRef<ReactElement | null>(null);
+  const handleOpenModal = (component: ReactElement) => {
```

Comment:
Usually, we use `handle` prefix with event handlers. In this case the function will be used not as event handler, but just like `handleOpenModal(<SomeComponent/>)`, inside of other functions. Maybe would be better to rename this function and `handleCloseModal` to just `openModal` and `closeModal`? What do you think?

Reply:
We decided to use a handle prefix for our handlers. It's a function that handles the opening modal. Also, it has been written in our code conventions 

![image](https://github.com/idx-academy/spa-orders/assets/118522791/147b1797-e87e-4be0-adad-874266b0553b)

Also provided example from the official MUI documentation
https://mui.com/material-ui/react-dialog/
![image](https://github.com/idx-academy/spa-orders/assets/118522791/41652190-78f4-4842-94c6-16ee67974d0f)

Reply:
I read our documentation before commenting. As for MUI examples, they pass those functions inside of event handlers inside of that one component - naming like this makes much sense. But in case of `useModalContext` - it is a reusable hook. In this example, `handleOpenAuthModal` name suits perfectly, but `handleOpenModal` makes it kinda confusing as for me: 
```
const handleOpenAuthModal = () => {
    handleOpenModal(<AuthModal />);
  };
```
As far as I know, `handle` prefix is set for functions which handle directly propagated by user actions like click on a button or scrolling. But there are cases when we need to open modal without user interaction, for example, inside of a `useEffect` when component is rendered for the first time. Then it is already not an action propagated by user. Even in cases, when we need to open modal on button click, we will have `handleButtonClick` or something like this, and inside we will programmatically call function to open modal.

Those are my thoughts. I don't want to be душним because of naming)

Reply:
Thank you for your explanation, it makes sense. I renamed the function according to your suggestion )